functions/problem-set-functions.md

```diff
@@ -0,0 +1,243 @@
+* language: javascript
+* id: a44b8bfc
+* title: Functions
+* docker_directory_path: /custom-snippets/isOdd
+##### !question
+
+For the fun of it, submit a solution to `isOdd` using an alternative syntax. Even if your original implementation is better, practice solving the same problem a different way!
+
+Consider trying:
+
+- Taking advantage of arrow function shortcut syntax
+- Making the function smaller
+- Making the function longer
+- Writing the conditional logic differently
+
+`isOdd` should take in one number, `num`. It returns a Truthy value if `num` is an odd number. It returns a Falsy value if `num` is an even number.
+
+##### !end-question
+### !explanation
+
+An example of a working implementation:
+
+```js
+const isOdd = num => {
+    return num % 2;
+};
+```
+
+Another example of a working implementation:
+
+```js
+const isOdd = num => num % 2;
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: custom-snippet
+* language: javascript
+* id: 0d7fc6c9
+* title: Functions
+* docker_directory_path: /custom-snippets/isOddNumber
+##### !question
+
+Implement the function `isOddNumber`.
+
+`isOddNumber` should take in one number, `num`. It returns a Truthy value if `num` is an odd number. It returns a Falsy value if `num` is an even number.
+
+If `num` is not a number, the function should return a Falsy value.
+
+| `num`   | Expected Return Value |
+| ------- | --------------------- |
+| `1`     | Truthy                |
+| `2`     | Falsy                 |
+| `hello` | Falsy                 |
+| `[]`    | Falsy                 |
+| `{}`    | Falsy                 |
+
+##### !end-question
+### !explanation
+
+- The result of `'hello' % 2` is `NaN`, which is falsy
+- The result of `[] % 2` is `0`, which is falsy
+- The result of `{} % 2` is `NaN`, which is falsy
+
+<br/>
+
+An example of a working implementation:
+
+```js
+const isOddNumber = num => {
+    return num % 2;
+};
+```
+
+Another example of a working implementation:
+
+```js
+const isOddNumber = num => num % 2;
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: custom-snippet
+* language: javascript
+* id: d522a65a
+* title: Functions
+* docker_directory_path: /custom-snippets/giveGreetingChallenge
+##### !question
+
+Implement the following three functions:
+
+- `giveCasualGreeting`
+- `giveFormalGreeting`
+- `writeLetter`
+
+`giveCasualGreeting` takes in one parameter, `name`.
+
+`name` | Expected Return Value
+--- | ---
+`'Miranda'` | `'Howdy, Miranda!'`
+`'Jonathan'` | `'Howdy, Jonathan!'`
+
+`givesFormalGreeting` takes in one parameter, `name`.
+
+`name` | Expected Return Value
+--- | ---
+`'Miranda'` | `'Dear Mx. Miranda 🤵'`
+`'Jonathan'` | `'Dear Mx. Jonathan 🤵'`
+
+`writeLetter` takes in _two_ parameters:
+- `giveGreeting`, which is a function
```

Comment:
(low priority)

This clashes with the simplified `name` that immediately follows, and may not be necessary due to the table of values, but I feel like practicing how to describe this parameter is useful.

```suggestion
- `giveGreeting`, which is a reference to function that accepts a single string parameter
```